akka-stream/src/main/scala/akka/stream/impl/ExposedPublisherReceive.scala

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import akka.actor.Actor
+
+/**
+ * INTERNAL API
+ */
+private[akka] abstract class ExposedPublisherReceive(activeReceive: Actor.Receive, unhandled: Any ⇒ Unit) extends Actor.Receive {
+  private var stash = List.empty[Any]
+
+  def isDefinedAt(o: Any): Boolean = true
+
+  def apply(o: Any): Unit = o match {
+    case ep: ExposedPublisher ⇒
+      receiveExposedPublisher(ep)
+      // we don't use sender() so this is allright
+      stash.reverse.foreach { msg ⇒
```

Comment:
I will add a `if (stash.nonEmpty)` here, just to make sure that there is no closure instance created for the normal case